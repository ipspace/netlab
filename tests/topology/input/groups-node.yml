defaults:
  device: cumulus
  
nodes:
  a:
    group: [ g2 ]
  b:
  c:
  d:
  e:
    group: [ g2, g3 ]
  f:
    group: g1 

groups:
  g1: [ a,b,c ]
  g2:
    vars:
      x1: 1
      x2: 2
  cumulus:
    vars:
      ansible_user: other
  all:
    vars:
      t1: 1
      t2: 2
