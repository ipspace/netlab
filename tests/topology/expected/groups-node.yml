groups:
  all:
    members: []
    vars:
      t1: 1
      t2: 2
  cumulus:
    members: []
    vars:
      ansible_user: other
  g1:
    members:
    - a
    - b
    - c
    - f
  g2:
    members:
    - a
    - e
    vars:
      x1: 1
      x2: 2
  g3:
    members:
    - e
input:
- topology/input/groups-node.yml
- package:topology-defaults.yml
name: input
nodes:
  a:
    af:
      ipv4: true
    box: CumulusCommunity/cumulus-vx:4.4.5
    device: cumulus
    group:
    - g2
    id: 1
    interfaces: []
    loopback:
      ifindex: 0
      ifname: lo
      ipv4: 10.0.0.1/32
      neighbors: []
      type: loopback
      virtual_interface: true
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.101
      mac: 08:4f:a9:00:00:01
    mtu: 1500
    name: a
  b:
    af:
      ipv4: true
    box: CumulusCommunity/cumulus-vx:4.4.5
    device: cumulus
    id: 2
    interfaces: []
    loopback:
      ifindex: 0
      ifname: lo
      ipv4: 10.0.0.2/32
      neighbors: []
      type: loopback
      virtual_interface: true
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.102
      mac: 08:4f:a9:00:00:02
    mtu: 1500
    name: b
  c:
    af:
      ipv4: true
    box: CumulusCommunity/cumulus-vx:4.4.5
    device: cumulus
    id: 3
    interfaces: []
    loopback:
      ifindex: 0
      ifname: lo
      ipv4: 10.0.0.3/32
      neighbors: []
      type: loopback
      virtual_interface: true
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.103
      mac: 08:4f:a9:00:00:03
    mtu: 1500
    name: c
  d:
    af:
      ipv4: true
    box: CumulusCommunity/cumulus-vx:4.4.5
    device: cumulus
    id: 4
    interfaces: []
    loopback:
      ifindex: 0
      ifname: lo
      ipv4: 10.0.0.4/32
      neighbors: []
      type: loopback
      virtual_interface: true
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.104
      mac: 08:4f:a9:00:00:04
    mtu: 1500
    name: d
  e:
    af:
      ipv4: true
    box: CumulusCommunity/cumulus-vx:4.4.5
    device: cumulus
    group:
    - g2
    - g3
    id: 5
    interfaces: []
    loopback:
      ifindex: 0
      ifname: lo
      ipv4: 10.0.0.5/32
      neighbors: []
      type: loopback
      virtual_interface: true
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.105
      mac: 08:4f:a9:00:00:05
    mtu: 1500
    name: e
  f:
    af:
      ipv4: true
    box: CumulusCommunity/cumulus-vx:4.4.5
    device: cumulus
    group:
    - g1
    id: 6
    interfaces: []
    loopback:
      ifindex: 0
      ifname: lo
      ipv4: 10.0.0.6/32
      neighbors: []
      type: loopback
      virtual_interface: true
    mgmt:
      ifname: eth0
      ipv4: 192.168.121.106
      mac: 08:4f:a9:00:00:06
    mtu: 1500
    name: f
provider: libvirt
